src/plugins/discover/public/application/components/default_discover_table/default_discover_table.tsx

```diff
@@ -50,8 +54,17 @@ export const LegacyDiscoverTable = ({
   onFilter,
   onClose,
   sampleSize,
+  showPagination,
 }: DefaultDiscoverTableProps) => {
+  const pageSize = 50;
```

Comment:
I would give anything for this to be changeable in Advanced Settings.

Reply:
i think what @ashwin-pc wants is that for now we just copied over how the legacy discover handles it in 2.9.

In 2.9, they just default set pagination for discover embeddable on dashboard page, and non-pagination for discover table on discover page. Also they just do 50 rows per page by default too. 

But yea i think we can add those settings in advanced setting in the future too.

Reply:
@AMoo-Miki In the new experience this is even easier to configure. The table footer has a drop down that the user can select to choose how many rows they want to display. For the legacy table lets keep the experience the same :)